flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -261,6 +261,12 @@
      */
     private volatile boolean isRunning;
 
+    /**
+     * Flag to mark the task "in operation", in which case check needs to be process the exceptions
+     * thrown in the initializing duration.
+     */
+    private volatile boolean isInitializing;
```

Comment:
Ah, I guess he just want to follow the `ExecutionState`'s initializing state, the corresponding java doc indicates that:
`Restoring last possible valid state of the task if it has it.`

Reply:
> Ah, I guess he just want to follow the `ExecutionState`'s initializing state, the corresponding java doc indicates that:

@akalash @reswqa yes. I described the section at the jira ticket. CC @1996fanrui

Reply:
Hm, I see, I need to think about that. 
Just in case, `StreamTask` and `ExecutionState` are different levels of abstraction and we don't have to keep them the same. We can do it but I'm not sure yet that it has any advantages. We should consider that `StreamTask` has two methods `init` and `restore` where `init` is part of `restore`. The `isInitializing` follows the `restore` but it says that it follows the `initialization` this  `initialization` is have nothing in common with `init` method but it relates to `ExecutionState` which relates to `Task` rather than `StreamTask`.
So I hope you understand now why I am confused.

Reply:
Thanks for the clarify.  I got it.
We need to define or describe different states for  different abstract level as accurately as possible, right?
I'd prefer your description `isRestoring`, now.

Reply:
> We need to define or describe different states for different abstract level as accurately as possible, right?

Yes, I think it is pretty much what I meant. We have a class/entity with a pretty clear lifecycle let's use these rather than trying to adapt to something else.
But of course, I'm happy to hear opinions from everybody.